rllib/evaluation/metrics.py

```diff
@@ -95,46 +92,39 @@ def collect_metrics(
 
 @DeveloperAPI
 def collect_episodes(
-    local_worker: Optional["RolloutWorker"] = None,
-    remote_workers: Optional[List[ActorHandle]] = None,
-    to_be_collected: Optional[List[ObjectRef]] = None,
+    workers: "WorkerSet",
+    remote_worker_ids: Optional[List[int]] = None,
     timeout_seconds: int = 180,
-) -> Tuple[List[RolloutMetrics], List[ObjectRef]]:
+) -> List[RolloutMetrics]:
     """Gathers new episodes metrics tuples from the given RolloutWorkers.
 
     Args:
-        local_worker: The local RolloutWorker (if any). By default, evaluation
-            WorkerSets don't have a local worker anymore (not needed).
-        remote_workers: List of ActorHandle pointing to remote RolloutWorkers.
+        workers: WorkerSet.
+        remote_worker_ids: Optional list of IDs of remote workers to collect
+            metrics from.
+        timeout_seconds: Timeout in seconds for collecting metrics from remote workers.
 
+    Returns:
+        List of RolloutMetrics.
     """
-    if remote_workers is None:
-        remote_workers = []
-
-    if to_be_collected is None:
-        to_be_collected = []
-
-    if remote_workers:
-        pending = [
-            a.apply.remote(lambda ev: ev.get_metrics()) for a in remote_workers
-        ] + to_be_collected
-        collected, to_be_collected = ray.wait(
-            pending, num_returns=len(pending), timeout=timeout_seconds * 1.0
-        )
-        if pending and len(collected) == 0:
-            logger.warning(
-                "WARNING: collected no metrics in {} seconds".format(timeout_seconds)
-            )
-        metric_lists = ray.get(collected)
-    else:
-        metric_lists = []
+    # This will drop get_metrics() calls that are too slow.
```

Comment:
The timeout is 180s, though, and I think it's hardcoded. Shouldn't be a problem, at least nothing should get dropped with that timeout :)

But yes! We should think about async'ing the metrics collection stuff.

Reply:
yep, yep.